controllers/overlaynetwork_controller.go

```diff
@@ -47,9 +53,18 @@ type OverlayNetworkReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
 func (r *OverlayNetworkReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	overlayNetwork := &selfservicev1.OverlayNetwork{}
+	if err := r.Client.Get(ctx, req.NamespacedName, overlayNetwork); err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to get OverlayNetwork %q: %v", req.NamespacedName, err)
+	}
 
-	// TODO(user): your logic here
+	netAttachDef := renderNetAttachDef(overlayNetwork)
```

Comment:
I think the code should try to fetch an already existing OverlayNetowrk CR with the same name.
If there is no existing CR, a new NAD should be created.
If the new and the old one are the same, nothing should be done.
If they are different an error should be returned (in the future we should think how we block OverlayNetwork updates in a nicer way).